cmd/jujud/agent/machine.go

```diff
@@ -264,22 +269,29 @@ func MachineAgentFactoryFn(
 func NewMachineAgent(
 	machineId string,
 	agentConfWriter AgentConfigWriter,
-	bufferedLogs logsender.LogRecordCh,
+	bufferedLogger *logsender.BufferedLogWriter,
 	runner worker.Runner,
 	loopDeviceManager looputil.LoopDeviceManager,
+	newIntrospectionSocketName func(names.Tag) string,
 	rootDir string,
 ) *MachineAgent {
-	return &MachineAgent{
+	a := &MachineAgent{
 		machineId:                   machineId,
 		AgentConfigWriter:           agentConfWriter,
 		configChangedVal:            voyeur.NewValue(true),
-		bufferedLogs:                bufferedLogs,
+		bufferedLogger:              bufferedLogger,
 		workersStarted:              make(chan struct{}),
 		runner:                      runner,
 		rootDir:                     rootDir,
 		initialUpgradeCheckComplete: gate.NewLock(),
 		loopDeviceManager:           loopDeviceManager,
+		newIntrospectionSocketName:  newIntrospectionSocketName,
+		prometheusRegistry:          newPrometheusRegistry(),
 	}
+	a.prometheusRegistry.MustRegister(
```

Comment:
Done. I might have argued against just because the panic is not possible with these statically defined collectors, but the error result will make it possible to add parameter validation to the constructors. So it's a good idea in that regard, and being able to check the errors from Register is a nice side benefit.